### PR TITLE
Separate containerd install from config, and other cleanups

### PR DESCRIPTION
**Pull request by pjh**
/kind cleanup

The current Windows GCE startup scripts mix installation and configuration steps in the same function for crictl and containerd. This is problematic because if we pre-install these binaries on the Windows nodes, then the current code will detect that they're installed and exit the function, skipping necessary configuration steps.

This PR separates installation from configuration, along with several other small cleanups.

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 1]
```diff
@@ -922,24 +907,60 @@ function Configure-GcePdTools {
     New-Item -path $PsHome\profile.ps1 -type file
   }
 
-    Add-Content $PsHome\profile.ps1 `
+  Add-Content $PsHome\profile.ps1 `
```
Comment (YangLu1031) [inline cluster/gce/windows/k8s-node-setup.psm1]: nit: also intent 3 lines below?

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 2]
```diff
@@ -1236,33 +1224,26 @@ function Verify-WorkerServices {
...
       -Algorithm SHA256
+}
 
-  Push-Location $tmp_dir
-  Move-Item -Force crictl.exe ${env:NODE_DIR}\
+# Sets crictl configuration values.
+function Configure-Crictl {
   if (${env:CONTAINER_RUNTIME_ENDPOINT}) {
-    crictl.exe config runtime-endpoint ${env:CONTAINER_RUNTIME_ENDPOINT}
+    & ${env:NODE_DIR}\crictl.exe config runtime-endpoint `
```
Comment (YangLu1031) [inline cluster/gce/windows/k8s-node-setup.psm1]: If ${env:NODE_DIR} path has spaces, should we quote the cmd path? eg. `& "${env:NODE_DIR}\crictl.exe"` https://ss64.com/ps/call.html

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 3]
```diff
@@ -58,6 +58,9 @@ $GCE_METADATA_SERVER = "169.254.169.254"
 # exist until an initial HNS network has been created on the Windows node - see
 # Add_InitialHnsNetwork().
 $MGMT_ADAPTER_NAME = "vEthernet (Ethernet*"
+$CNI_RELEASE_VERSION = 'v0.8.5'
```
Comment (yliaog) [inline cluster/gce/windows/k8s-node-setup.psm1]: why is this needed?

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 4]
```diff
@@ -296,29 +300,11 @@ function Create-Directories {
...
-  # -match does regular expression matching.
-  if ($WinVersion -match '1809') {
-    return '1809'
-  }
-  elseif ($WinVersion -match '2019') {
-    return 'ltsc2019'
+  if (ShouldWrite-File ${env:K8S_DIR}\hns.psm1) {
+    MustDownload-File `
+        -OutFile ${env:K8S_DIR}\hns.psm1 `
+        -URLs "https://www.googleapis.com/storage/v1/b/gke-release/o/winnode%2fconfig%2fsdn%2fmaster%2fhns.psm1?alt=media"
```
Comment (yliaog) [inline cluster/gce/windows/k8s-node-setup.psm1]: better to use https://storage.googleapis.com/gke-release/winnode/config/sdn/master/hns.psm1

File: cluster/gce/windows/k8s-node-setup.psm1 [hunk 5]
```diff
@@ -922,24 +907,60 @@ function Configure-GcePdTools {
...
+  if (-not (ShouldWrite-File ${env:CNI_DIR}\win-bridge.exe) -and
+      -not (ShouldWrite-File ${env:CNI_DIR}\host-local.exe)) {
+    return
+  }
+
+  $tmp_dir = 'C:\cni_tmp'
+  New-Item $tmp_dir -ItemType 'directory' -Force | Out-Null
+
+  $release_url = ('https://storage.googleapis.com/k8s-artifacts-cni/' +
+      'release%2f' + $CNI_RELEASE_VERSION + '%2f')
```
Comment (yliaog) [inline cluster/gce/windows/k8s-node-setup.psm1]: i think %2f can be replaced by '/'

Comment (yliaog) [reply cluster/gce/windows/k8s-node-setup.psm1]: also this is changed to env variable, it needs to be merge/resolve

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 3 above)
Comment (pjh) [reply cluster/gce/windows/k8s-node-setup.psm1]: This was just moving a constant to the top of the file, but this is now obsolete after rebasing.

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 4 above)
Comment (pjh) [reply cluster/gce/windows/k8s-node-setup.psm1]: Done.

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 1 above)
Comment (pjh) [reply cluster/gce/windows/k8s-node-setup.psm1]: Indenting the lines below would indent the contents of the file, which we don't want.

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 2 above)
Comment (pjh) [reply cluster/gce/windows/k8s-node-setup.psm1]: Good point, done.

File: cluster/gce/windows/k8s-node-setup.psm1 (hunk 5 above)
Comment (pjh) [reply cluster/gce/windows/k8s-node-setup.psm1]: Rebased + resolved.

**Comment by pjh** [comment]
@yliaog @YangLu1031 thanks for the comments, PTAL.

**Comment by pjh** [comment]
@yliaog PTAL, thanks
